boilerplate/fullstack/remix/app/routes/_index.tsx

```diff
@@ -0,0 +1,165 @@
+import { CelebrateIcon, SeparatorLine, BlogsIcon, GuideIcon, SignOutIcon } from "../../assets/images";
+import { recipeDetails } from "../config/frontend";
+import SuperTokens from "supertokens-auth-react";
+import { LoaderFunctionArgs, redirect } from "@remix-run/node";
+import { useLoaderData } from "@remix-run/react";
+import SessionReact from "supertokens-auth-react/recipe/session/index.js";
+import { getSessionDetails } from "../lib/superTokensHelpers";
+import { TryRefreshComponent } from "../components/tryRefreshClientComponent";
+import { SessionAuthForRemix } from "../components/sessionAuthForRemix";
+
+interface SessionDataForUI {
+    note: string;
+    userId: string;
+    sessionHandle: string;
+    accessTokenPayload: object;
+}
+
+interface SessionForRemixProps {
+    userId: string;
+    sessionHandle?: string;
+    accessTokenPayload: SessionDataForUI;
+}
+
+export async function loader({ request }: LoaderFunctionArgs): Promise<{
+    session: SessionForRemixProps | undefined;
+    hasInvalidClaims: boolean;
+    hasToken: boolean;
+    RemixResponse: Response | null;
+}> {
+    const { session, hasInvalidClaims, hasToken, RemixResponse } = await getSessionDetails(request);
+
+    const res: SessionForRemixProps = {
+        userId: session?.getUserId(),
+        sessionHandle: session?.getHandle(),
+        accessTokenPayload: session?.getAccessTokenPayload(),
+    };
+
+    if (RemixResponse) {
+        return {
+            session: res,
+            hasInvalidClaims,
+            hasToken,
+            RemixResponse,
+        };
+    } else {
+        return {
+            session: res,
+            hasInvalidClaims,
+            hasToken,
+            RemixResponse: null,
+        };
+    }
+}
+
+export default function Home() {
+    const loaderData = useLoaderData<{
+        session: SessionForRemixProps | undefined;
+        hasInvalidClaims: boolean;
+        hasToken: boolean;
+        RemixResponse: Response | null;
+    }>();
+
+    console.log(loaderData);
+    console.log(loaderData.session);
+
+    async function logoutClicked() {
+        await SessionReact.signOut();
+        SuperTokens.redirectToAuth();
+    }
+
+    if (!loaderData.session) {
+        if (!loaderData.hasToken) {
+            return redirect("/auth");
+        }
+        if (loaderData.hasInvalidClaims) {
+            return <SessionAuthForRemix />;
+        } else {
+            return <TryRefreshComponent />;
+        }
+    }
+    if (loaderData.session) {
+        const sessionData: SessionDataForUI = {
+            note: "Retrieve authenticated user-specific data from your application post-verification through the use of the verifySession middleware.",
+            userId: loaderData.session?.userId,
```

Comment:
you should not need to do `loaderData.session?` cause we check above that if it's `if (!loaderdata.session)`. Sometimes TS can't infer this on its own, so you should assign `loaderData.session` into a variable and then use that in the function.

Reply:
✔️